java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/RuntimeClassPathUtils.java

```diff
@@ -76,4 +93,87 @@
         }
         return classPathList.toArray(new String[classPathList.size()]);
     }
+
+    private static String[] resolveClasspath(IJavaProject javaProject) throws CoreException {
+//        final JUnitLaunchConfigurationDelegate delegate = new JUnitLaunchConfigurationDelegate();
+        final ILaunchConfiguration launchConfig = new JavaApplicationLaunchConfiguration(javaProject.getProject());
+//        delegate.getClasspath(launchConfig);
+        final IRuntimeClasspathEntry[] unresolved = JavaRuntime.computeUnresolvedRuntimeClasspath(launchConfig);
+        final IRuntimeClasspathEntry[] resolved = JavaRuntime.resolveRuntimeClasspath(unresolved, launchConfig);
+        final Set<String> classpaths = new LinkedHashSet<>();
+        for (final IRuntimeClasspathEntry entry : resolved) {
+            final String location = entry.getLocation();
+            if (location != null) {
+                if (entry.getClasspathProperty() == IRuntimeClasspathEntry.USER_CLASSES ||
+                        entry.getClasspathProperty() == IRuntimeClasspathEntry.CLASS_PATH) {
+                    classpaths.add(location);
+                }
+                // TODO: support module path
+            }
+        }
+        return classpaths.toArray(new String[classpaths.size()]);
+    }
+
+    private static class JavaApplicationLaunchConfiguration extends LaunchConfiguration {
+        public static final String JAVA_APPLICATION_LAUNCH = "<?xml version=\"1.0\" encoding=\"UTF-8\"" +
+                " standalone=\"no\"?>\n" +
+                "<launchConfiguration type=\"org.eclipse.jdt.junit.launchconfig\">\n" +
+                "<listAttribute key=\"org.eclipse.debug.core.MAPPED_RESOURCE_PATHS\">\n" +
+                "</listAttribute>\n" +
+                "<listAttribute key=\"org.eclipse.debug.core.MAPPED_RESOURCE_TYPES\">\n" +
+                "<listEntry value=\"1\"/>\n" +
+                "</listAttribute>\n" +
+                "<booleanAttribute key=\"org.eclipse.jdt.launching.ATTR_EXCLUDE_TEST_CODE\" value=\"false\"/>\n" +
+                "</launchConfiguration>";
+        private final IProject project;
+        private String classpathProvider;
+        private String sourcepathProvider;
+        private final LaunchConfigurationInfo launchInfo;
+
+        protected JavaApplicationLaunchConfiguration(IProject project) throws CoreException {
+            super(String.valueOf(new Date().getTime()), null, false);
+            this.project = project;
+            if (ProjectUtils.isMavenProject(project)) {
+                classpathProvider = "org.eclipse.m2e.launchconfig.classpathProvider";
+                sourcepathProvider = "org.eclipse.m2e.launchconfig.sourcepathProvider";
+            } else if (ProjectUtils.isGradleProject(project)) {
+                classpathProvider = "org.eclipse.buildship.core.classpathprovider";
+            }
```

Comment:
what if not maven or gradle?

Reply:
Works as expected.

![demo](https://user-images.githubusercontent.com/6193897/60071938-705e9200-974e-11e9-9312-a5ed2822e0da.gif)